### PR TITLE
CB-15255 SDX flow trigger events should implement IdempotentEvent int…

### PR DESCRIPTION
**Pull request by lturcsanyi**
…erface

See detailed description in the commit message.

### CODE REVIEW

File: datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeFlowChainStartEvent.java [hunk 1]
```diff
@@ -36,4 +38,12 @@ public String getBackupLocation() {
     public String selector() {
         return SDX_RESIZE_FLOW_CHAIN_START_EVENT;
     }
+
+    @Override
+    public boolean equalsEvent(SdxEvent other) {
+        return isClassAndEqualsEvent(DatalakeResizeFlowChainStartEvent.class, other,
+                event -> Objects.equals(sdxCluster, event.sdxCluster)
```
Comment (lnardai) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeFlowChainStartEvent.java:45]: Wouldn't it be better to only check if the CRN is the same?

**Review by foldik — APPROVED** [review]
LGTM

File: datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeFlowChainStartEvent.java (hunk 1 above)
Comment (lnardai) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeFlowChainStartEvent.java:45]: Nevermind, it's the new sdxCluster that is replacingTheOther? It has a misleading name in the model then.

**Review by lnardai — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeFlowChainStartEvent.java (hunk 1 above)
Comment (lturcsanyi) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeFlowChainStartEvent.java:45]: Yes, its the new sdxCluster.

Comment (lturcsanyi) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeFlowChainStartEvent.java:45]: The equals of SdxCluster uses the created timestamp which always make equals return false. I will fix this.

**Pull request merged**